### DISCUSSION TITLE
An error when pushing invoice with tax rate Id

### DISCUSSION THREAD

**Discussion by mcclowes** [codatio/codat-docs#118] [Q&A] [answered]
_Migrated question from old forum_

In our sandbox we set up codat to connect to a fresh Quickbook Online instance.
We export invoices to codat and we have TaxRateRef.Id = 2_Invoices which comes from Quickbooks Online. We check by fetching via Codat with swagger and got the correct tax rate. But whenever we export an invoice with tax then it will throw that error even if that should be the valid tax rate id. The export seems to work with NON-tax.
It seems to work last year, but for now, it doesn't work anymore.

Does it relate to this post: https://docs.codat.io/changelog/36186-us-tax-handling?

Or do you have any idea what was wrong with that? This is an example payload we have when we push invoices to codat

```
"lineItems": [
{
"description": "Example Product",
"unitAmount": 80.0000000000,
"quantity": 1.0000000000,
"subTotal": 80.00,
"taxAmount": 0,
"totalAmount": 80.00,
"accountRef": {
"id": "1",
"name": "Services"
},
"taxRateRef": {
"id": "2_Invoices",
"name": "California",
"effectiveTaxRate": 0
},
"itemRef": {
"id": "1",
"name": "Services"
},
"trackingCategoryRefs": [],
"isDirectIncome": false
}
],
```

**Comment by mcclowes** [comment]
I have a quick question regarding your Quickbooks Online account.

Is the Automated Sales Tax (AST) feature enabled on your account (if using the US version of QBO)? If so, it would be as simple as pushing the taxRateRef.id "TAX".

Allowing QuickBooks Online US to handle the sales tax based on location etc

**Reply by cuongtran-6618** [reply]
Hi Clayton, thank you for your answer. The issue is that the tax rate of QBO and Codat did not sync fully. We create several tax rates in the QBO site, and after syncing again to Codat, it can be fetched via swagger. The problem is when we want to push the invoice there, Codat said that the tax rate is not found even though it was there in the swagger response. One thing we noticed

` "validDatatypeLinks": []. `

This is a different with 2 default tax rate "TAX" and "NON". which looks like that 

```
"validDatatypeLinks": [
{
"property": "Id",
"links": [
"Invoice.LineItems.TaxRateRef.Id",
"Bill.LineItems.TaxRateRef.Id",
"BillCreditNote.LineItems.TaxRateRef.Id"
]
}
]
```

We would like to push that with the tax rate we defined rather than let AST pick up so that would be great if we can figure out why the tax rate is denied by Codat's side.

Best regard,

Cuong

**Reply by Ojo-Joseph** [reply]
Hi Cuong,

I want to let you know that the issue you mentioned is a known bug, and it's already on our backlog to be fixed. Before, we only supported pushing invoices with AST enabled, but we're working on expanding our support for user-defined rates.

Kind Regards,

Joseph

**Reply by cuongtran-6618** [reply]
Thank a lot for clear it out Joseph.

**Reply by Ojo-Joseph** [reply, marked as answer]
Hey Cuong,

I wanted to bring to your attention that the issue you reported earlier has been resolved. As a result, it is now possible for you to push custom tax rates even when AST (Automated Sales Tax) is enabled.

Kind regards,

Joseph
